### PR TITLE
nimble/audio/bass: BASS modify source update

### PR DESCRIPTION
**Pull request by szymon-czapracki**
Update modify source to work with
new mbuf-style data input.

Also a handling for proper receive state fields is added.

Fix a never-ending goto statement returning to goto itself.

### CODE REVIEW

File: nimble/host/audio/services/bass/include/services/bass/ble_audio_svc_bass.h [hunk 1]
```diff
@@ -325,7 +325,11 @@ struct ble_svc_audio_bass_operation {
             uint16_t num_subgroups;
 
             /** BIS Synchronisation of subgroups */
+            M
```
Comment (rymanluk) [inline nimble/host/audio/services/bass/include/services/bass/ble_audio_svc_bass.h:328]: looks like trash :)

File: nimble/host/audio/services/bass/src/ble_audio_svc_bass.c [hunk 2]
```diff
@@ -477,8 +479,7 @@ ble_svc_audio_bass_modify_source(uint8_t *data, uint16_t data_len, uint16_t conn
 
     operation.modify_source.pa_interval = get_le16(&data[offset]);
     offset += 2;
-    operation.modify_source.num_subgroups = get_le16(&data[offset]);
-    offset += 2;
+    operation.modify_source.num_subgroups = get_le16(&data[offset++]);
```
Comment (rymanluk) [inline nimble/host/audio/services/bass/src/ble_audio_svc_bass.c:482]: is `offset++`  intended here?

Comment (szymon-czapracki) [reply nimble/host/audio/services/bass/src/ble_audio_svc_bass.c]: It's moved below - this was a mistake. We have to move offset after reading number of subroups (which size is 1 byte - but I'm sure u know that :) ) 
image

File: nimble/host/audio/services/bass/include/services/bass/ble_audio_svc_bass.h (hunk 1 above)
Comment (szymon-czapracki) [reply nimble/host/audio/services/bass/include/services/bass/ble_audio_svc_bass.h]: Removed

File: nimble/host/audio/services/bass/src/ble_audio_svc_bass.c [hunk 3]
```diff
@@ -478,7 +479,7 @@ ble_svc_audio_bass_modify_source(uint8_t *data, uint16_t data_len, uint16_t conn
     operation.modify_source.pa_interval = get_le16(&data[offset]);
     offset += 2;
     operation.modify_source.num_subgroups = get_le16(&data[offset]);
-    offset += 2;
```
Comment (rymanluk) [inline nimble/host/audio/services/bass/src/ble_audio_svc_bass.c:481]: this seems to be bad

File: nimble/host/audio/services/bass/src/ble_audio_svc_bass.c [hunk 4]
```diff
@@ -489,7 +490,18 @@ ble_svc_audio_bass_modify_source(uint8_t *data, uint16_t data_len, uint16_t conn
 
     for (i = 0; i < operation.modify_source.num_subgroups; i++) {
         operation.modify_source.bis_sync[i] = get_le32(&data[offset]);
+        rcv_state->state.subgroups[i].bis_sync_state = operation.modify_source.bis_sync[i];
```
Comment (rymanluk) [inline nimble/host/audio/services/bass/src/ble_audio_svc_bass.c:493]: this state will be updated when actual SYNC will happen. At this stage we cannot do that as we overwrite a state.

File: nimble/host/audio/services/bass/src/ble_audio_svc_bass.c [hunk 5]
```diff
@@ -509,12 +521,30 @@ ble_svc_audio_bass_modify_source(uint8_t *data, uint16_t data_len, uint16_t conn
...
+    }
 
 done:
     if (!rc) {
         rc = ble_svc_audio_bass_receive_state_notify(rcv_state);
         ev.bass_operation_status.status = rc;
-        goto done;
     }
 
     ble_audio_event_listener_call(&ev);
```
Comment (rymanluk) [inline nimble/host/audio/services/bass/src/ble_audio_svc_bass.c:550]: this even here, will request application to sync. Once this is done, state will be set correctly and expected notification will be sent out

File: nimble/host/audio/services/bass/src/ble_audio_svc_bass.c (hunk 4 above)
Comment (szymon-czapracki) [reply nimble/host/audio/services/bass/src/ble_audio_svc_bass.c]: Removed

**Pull request merged**